### PR TITLE
JST with matrix dissipation

### PR DESCRIPTION
**Pull request by pcarruscag**
## Proposed Changes
JST with scalar dissipation over predicts drag in friction-dominated problems, the version with matrix dissipation is better.

## Related Work
Depends on #1022

## PR Checklist
- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [X] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: SU2_CFD/include/numerics_simd/flow/convection/centered.hpp [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+    }
+  }
+};
+
+/*!
+ * \class CJSTmatScheme
+ * \brief JST scheme with matrix dissipation.
+ */
+template<class Decorator>
+class CJSTmatScheme : public CCenteredBase<CJSTmatScheme<Decorator>,Decorator> {
```
Comment (pcarruscag) [inline SU2_CFD/include/numerics_simd/flow/convection/centered.hpp:272]: Implementation starts here.

**Comment by aeroamit** [comment]
Hi @pcarruscag,
Is this matrix dissipation version is going to be as robust as original JST in use (which does well with multigrid, while other low dissipation schemes do not i. e. Roe, HLLC, AUSM etc.)

**Comment by pcarruscag** [comment]
I have not tried it on anything very challenging yet, but I expect the usual trade-off between accuracy and robustness.
Do note that with enough entropy correction this version becomes nearly equivalent to normal JST (which will still be available).

**Comment by pcarruscag** [comment]
Merging this with #1022

**Pull request closed without merging**